src/app/lists/ListDataview.tsx

```diff
@@ -133,9 +134,28 @@ const ListsDataView: React.FC<Props> = (props) => {
             throw new DatabaseError("update", "lists items");
         }
 
-        const temp = row1.rowOrder;
-        row1.rowOrder = row2.rowOrder;
-        row2.rowOrder = temp;
+        const primaryKeys = listOfIngridients.map(
+            (listOfIngridients) => listOfIngridients.primary_key
+        );
+
+        const index1 = primaryKeys.indexOf(row1.primaryKey);
+        const index2 = primaryKeys.indexOf(row2.primaryKey);
+
+        const item1 = listOfIngridients[index1];
+        const order1 = item1.row_order;
+
+        const item2 = listOfIngridients[index2];
+        const order2 = item2.row_order;
+
+        item1.row_order = order2;
+        item2.row_order = order1;
+
+        const newData = [...listOfIngridients];
+
+        newData[index1] = item2;
+        newData[index2] = item1;
+
+        setData(newData);
```

Comment:
maybe worth putting this in a separate function as discussed :)

Reply:
Moved this into a separate function